### DISCUSSION TITLE
Error while Installing DeepMD-kit C++ Interface.

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#1760] [Q&A] [answered]
[ 85%] Linking CXX executable dp_ipi_low
/usr/bin/ld: dp_ipi_low: hidden symbol `cudaMemcpy' in /usr/local/cuda-11.7/lib64/libcudart_static.a(cudart_static.o) is referenced by DSO
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [ipi/CMakeFiles/dp_ipi_low.dir/build.make:105: ipi/dp_ipi_low] Error 1
make[1]: *** [CMakeFiles/Makefile2:499: ipi/CMakeFiles/dp_ipi_low.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 86%] Building CXX object op/CMakeFiles/deepmd_op.dir/prod_force.cc.o
[ 87%] Linking CXX executable dp_ipi
[ 88%] Building CXX object op/CMakeFiles/deepmd_op.dir/prod_force_multi_device.cc.o
/usr/bin/ld: dp_ipi: hidden symbol `cudaMemcpy' in /usr/local/cuda-11.7/lib64/libcudart_static.a(cudart_static.o) is referenced by DSO
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [ipi/CMakeFiles/dp_ipi.dir/build.make:105: ipi/dp_ipi] Error 1
make[1]: *** [CMakeFiles/Makefile2:444: ipi/CMakeFiles/dp_ipi.dir/all] Error 2
[ 90%] Building CXX object op/CMakeFiles/deepmd_op.dir/prod_virial.cc.o
[ 91%] Linking CXX shared library libdeepmd_gromacs.so
[ 91%] Built target deepmd_gromacs
[ 92%] Building CXX object op/CMakeFiles/deepmd_op.dir/prod_virial_multi_device.cc.o
[ 93%] Building CXX object op/CMakeFiles/deepmd_op.dir/soft_min.cc.o
[ 95%] Building CXX object op/CMakeFiles/deepmd_op.dir/soft_min_force.cc.o
[ 96%] Building CXX object op/CMakeFiles/deepmd_op.dir/soft_min_virial.cc.o
[ 97%] Building CXX object op/CMakeFiles/deepmd_op.dir/tabulate_multi_device.cc.o
[ 98%] Building CXX object op/CMakeFiles/deepmd_op.dir/unaggregated_grad.cc.o
[100%] Linking CXX shared module libdeepmd_op.so
[100%] Built target deepmd_op
make: *** [Makefile:136: all] Error 2

**Comment by njzjz** [comment, marked as answer]
This error was fixed by #1731.

**Reply by AnuragKr** [reply]
I am still getting this error after installing it again via source code.
Cmake Command - make -DTENSORFLOW_ROOT=$tensorflow_root -DUSE_CUDA_TOOLKIT=TRUE -DCMAKE_INSTALL_PREFIX=$deepmd_root ..
I got one solution telling that use cuda_add_library rather target_link_libraries in the CMakeLists.txt. But I couldn't find target_link_libraries in that file.(https://forums.developer.nvidia.com/t/make-error-hidden-symbol-cudamemcpyasync/112660)

**Reply by AnuragKr** [reply]
I fixed the issue by looking at the pull request and making appropriate changes to my local file. 
Thanks for your prompt response.
